src/java.base/share/classes/java/nio/charset/Charset.java

```diff
@@ -648,12 +649,19 @@ public SortedMap<String,Charset> run() {
      *
      * @since 1.5
      */
+    @SuppressWarnings("removal")
     public static Charset defaultCharset() {
         if (defaultCharset == null) {
             synchronized (Charset.class) {
-                String csn = GetPropertyAction
-                        .privilegedGetProperty("file.encoding");
-                Charset cs = lookup(csn);
+                // do not look for providers other than the standard one
+                Charset cs;
+                if (System.getSecurityManager() == null) {
+                    cs = standardProvider.charsetForName(StaticProperty.fileEncoding());
+                } else {
+                    PrivilegedAction<Charset> pa =
+                        () -> standardProvider.charsetForName(StaticProperty.fileEncoding());
```

Comment:
what operation does this do security permission check?  I don't think any.

Reply:
Right. I was under the impression that the call should have been made with privileged action (as in the comments for those access methods in `StaticProperty`, but the property value itself is not given to untrusted code. Will remove the permission check.